crates/jit-icache-coherence/src/libc.rs

```diff
@@ -64,6 +64,47 @@ pub(crate) fn pipeline_flush_mt() -> Result<()> {
     Ok(())
 }
 
+#[cfg(all(target_arch = "riscv64", target_os = "linux"))]
+fn riscv_flush_icache(start: u64, end: u64) -> Result<()> {
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "one-core")] {
+            use std::arch::asm;
+            unsafe {
+                asm!("fence.i");
+            };
+            return Ok(())
```

Comment:
`return` isn't necessary here, I think: the whole `if` is an expression that will return the last value in the block.